public/app/core/components/TransformersUI/CalculateFieldTransformerEditor.tsx

```diff
@@ -324,6 +354,66 @@ export class CalculateFieldTransformerEditor extends React.PureComponent<
     );
   }
 
+  //---------------------------------------------------------
+  // Format date
+  //---------------------------------------------------------
+
+  updateDateOptions = (opts: DateOptions) => {
+    const { options, onChange } = this.props;
+    onChange({
+      ...options,
+      mode: CalculateFieldMode.FormatDate,
+      date: opts,
+    });
+  };
+
+  onDateFormatChanged = (evt: ChangeEvent<HTMLInputElement>) => {
+    const { date } = this.props.options;
+    this.updateDateOptions({
+      ...date!,
+      format: evt.target.value,
+    });
+  };
+
+  onFieldToFormatChanged = (v: SelectableValue<string>) => {
+    const { date } = this.props.options;
+    this.updateDateOptions({
+      ...date!,
+      fieldName: v.value! as BinaryOperationID,
+    });
+  };
+
+  renderFormatDate(options?: DateOptions) {
+    const { timeFieldNames } = this.state;
+    const timeFieldOptions = [...timeFieldNames].map(fn => ({ label: fn, value: fn }));
+
+    return (
+      <div className="gf-form-inline">
+        <div className="gf-form">
+          <InlineLabel className="width-8">Field to format</InlineLabel>
+          <Select
+            isClearable={false}
+            allowCustomValue={false}
+            className="width-18"
+            options={timeFieldOptions}
+            placeholder={'Select field'}
+            value={options?.fieldName ?? ''}
+            onChange={this.onFieldToFormatChanged}
+          />
+          <InlineLabel width="auto" tooltip={'Look at the Moment.js docs for allowed formats'}>
```

Comment:
I suggest, rather than referring to moment.js docs, to have the allowed tokens listed or linked to in this particular transformation's docs. Or ideally if the allowed tokens were listed in this inline help. Problem though is it's gonna be fairly long list.